app/(candidate)/dashboard/questions/components/FunFact.js

```diff
@@ -9,19 +9,12 @@ export default function FunFact({
   value,
   onChangeCallback,
   saveCallback,
-  campaign,
   campaignKey,
 }) {
   const handleSave = () => {
     if (!canSave()) return;
-    const updated = {
-      ...campaign,
-      details: {
-        ...campaign.details,
-        [campaignKey]: value,
-      },
-    };
-    saveCallback(updated);
+
+    saveCallback([`details.${campaignKey}`], [value]);
   };
 
   const canSave = () => {
```

Comment:
wow... these are all three the same methods 😳 

Can we consolidate somehow?

<img width="1113" alt="image" src="https://github.com/thegoodparty/gp-webapp/assets/496983/f05ce8c9-1d19-4ca0-ae43-8c990f762ef5">